src/Firestore/Transaction.php

```diff
@@ -112,6 +112,60 @@ public function snapshot(DocumentReference $document, array $options = [])
         ] + $options);
     }
 
+    /**
+     * Get a list of documents by their path.
+     *
+     * The number of results generated will be equal to the number of documents
+     * requested, except in case of error.
+     *
+     * Note that this method will **always** return instances of
+     * {@see Google\Cloud\Firestore\DocumentSnapshot}, even if the documents
+     * requested do not exist. It is highly recommended that you check for
+     * existence before accessing document data.
```

Comment:
What happens if a user doesn't do this?

Reply:
If the document doesn't exist, `DocumentSnapshot::fields()` would return an empty array, and array-access (i.e. `$snapshot['field']`) would raise a PHP notice.